transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueueSocketChannel.java

```diff
@@ -19,45 +19,28 @@
 import io.netty.channel.socket.ServerSocketChannel;
 import io.netty.channel.socket.SocketChannel;
 import io.netty.util.concurrent.GlobalEventExecutor;
-import io.netty.util.internal.PlatformDependent;
 import io.netty.util.internal.UnstableApi;
 
-import java.net.InetAddress;
 import java.net.InetSocketAddress;
-import java.net.SocketAddress;
-import java.net.UnknownHostException;
-import java.nio.channels.AlreadyConnectedException;
 import java.util.concurrent.Executor;
 
 @UnstableApi
 public final class KQueueSocketChannel extends AbstractKQueueStreamChannel implements SocketChannel {
     private final KQueueSocketChannelConfig config;
 
-    private volatile InetSocketAddress local;
-    private volatile InetSocketAddress remote;
-    private InetSocketAddress requestedRemote;
-
     public KQueueSocketChannel() {
         super(null, BsdSocket.newSocketStream(), false);
         config = new KQueueSocketChannelConfig(this);
     }
 
     public KQueueSocketChannel(int fd) {
         super(new BsdSocket(fd));
-        // As we create an EpollSocketChannel from a FileDescriptor we should try to obtain the remote and local
-        // address from it. This is needed as the FileDescriptor may be bound/connected already.
-        remote = socket.remoteAddress();
-        local = socket.localAddress();
         config = new KQueueSocketChannelConfig(this);
     }
 
-    KQueueSocketChannel(Channel parent, BsdSocket fd, InetSocketAddress remote) {
-        super(parent, fd, true);
+    KQueueSocketChannel(Channel parent, BsdSocket fd, InetSocketAddress remoteAddress) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1135)